### DISCUSSION TITLE
Welcome to home_assistant_solarman Discussions!

### DISCUSSION THREAD

**Discussion by StephanJoubert** [StephanJoubert/home_assistant_solarman#51] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by cyberrep** [comment]
thats a cool feature!
First of all congrats Stephan to write this component to us, probably some
of us have experience with programming and others not.

I'm Rodrigo from Brazil 41 years old, graduated in computer science from at
2002, but I never work with computer, I ever work for finance department
and now as a Project Manager!
I am an enthusiastic of new tech, I installed HA in the beginning of the
pandemic to do something different, now I have some stuffs on my house I
can control!



Em qui., 26 de mai. de 2022 às 05:58, StephanJoubert <
***@***.***> escreveu:

> 👋 Welcome!
>
> We’re using Discussions as a place to connect with other members of our
> community. We hope that you:
>
>    - Ask questions you’re wondering about.
>    - Share ideas.
>    - Engage with other community members.
>    - Welcome others and are open-minded. Remember that this is a
>    community we
>    build together 💪.
>
> To get started, comment below with an introduction of yourself and tell us
> about what you do with this community.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/StephanJoubert/home_assistant_solarman/discussions/51>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFSCRIZDFDDYHAN52YVAJKTVL44M3ANCNFSM5XALGSJA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 


Rodrigo Tadeu Pasetchny - PMP
Tel + 55 11 991-789-268
USA + 1 859 721 14-44

**Comment by gedger** [comment]
Agreed, this is a much better way rather than raising issues....Also thanks to Stephen for creating the integration.

I'm Richard from Norfolk in the UK. I'm now retired but an ex. researcher and developer, I started coding in assembly on the Intel 4040 so you can guess my age, but also good with hardware. I spent a happy few hours decoding the RS485 commands for my Solis Inverter and posted it up, I'm convinced there are more parameters to find but it works well enough for me. Happy to try and help anyone else who wants to decode any new inverters.

All the best.

**Comment by cyberrep** [comment]
Nice to hear Richard!!!

I'm blowing my mind trying to understand modbus instructions to fit my
inverter at the configuration yaml kkkkk ,

Em qui., 26 de mai. de 2022 às 08:27, gedger ***@***.***>
escreveu:

> Agreed, this is a much better way rather than raising issues....Also
> thanks to Stephen for creating the integration.
>
> I'm Richard from Norfolk in the UK. I'm now retired but an ex. researcher
> and developer, I started coding in assembly on the Intel 4040 so you can
> guess my age, but also good with hardware. I spent a happy few hours
> decoding the RS485 commands for my Solis Inverter and posted it up, I'm
> convinced there are more parameters to find but it works well enough for
> me. Happy to try and help anyone else who wants to decode any new inverters.
>
> All the best.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/StephanJoubert/home_assistant_solarman/discussions/51#discussioncomment-2827025>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFSCRI7NFJHU25Z6JPBZTCLVL5NZBANCNFSM5XALGSJA>
> .
> You are receiving this because you commented.Message ID:
> <StephanJoubert/home_assistant_solarman/repo-discussions/51/comments/2827025
> @github.com>
>


-- 


Rodrigo Tadeu Pasetchny - PMP
Tel + 55 11 991-789-268
USA + 1 859 721 14-44

**Reply by wanderson-vicente** [reply]
Hello, my name is Wanderson I'm just a hobbyist and I'm starting in the HA universe. I'm trying to put the solarman integration, but when I search I can't find it in my HA.

**Reply by wanderson-vicente** [reply]
Partially solved, I installed it manually, but I can't connect, through the browser when I access DEYE I have to put the user "admin" and password "admin" in the integration I didn't find where I enter with this data. Can you help me?

**Reply by gedger** [reply]
[http192.168.1.40config_hide.html.pdf](https://github.com/StephanJoubert/home_assistant_solarman/files/8818488/http192.168.1.40config_hide.html.pdf)
The integration doesn't use the web interface but through a solarman5 interface so doesn't need the password. Too see if your comms stick is compatible try going to the hidden url.

http:\YOUR_INVERTER_IP\config_hide.html

There should be a line mentioning a port open on 8899, my inverter is asleep so I can't show an example. Not all comms sticks support this option.

edit: attached a copy.

**Reply by wanderson-vicente** [reply]
The Internal server parameters setting part is exactly like the PDF, but it doesn't.

Thank you for your help.

**Comment by OriginalPentAr** [comment]
Hi Stephan, thanks for the integration. I have an older Trannergy inverter (serial starts with 61xxx), which supports the SolarMan app. I added IP and serial, unfortunately it is not collecting any data. I was wondering, where do I find version 4/5 of the communication protocol? Thanks!

**Reply by pjannink** [reply]
I have the same question for a Trannergy PCL2300K. Would love to see the integration work for that model. Solarman app is working fine for it.

**Comment by deanboltman** [comment]
Hi all, the integration works perfectly on my Deye 5kW inverter. In the data graphs for the entities, there are gaps in the data showing, for example, SOC detail where it dips to 0. I've attached an example of the graph. It'll show these gaps in data on all entities. Is it due to the refresh data interval?  If so, how do I set it different refresh rate?
![Uploading Screenshot_20220703-055412_Home Assistant.jpg…]()

**Comment by asiviero76** [comment]
All, anyone has tried to integrate the inverter HYD 3000-ZSS HP of ZCS zucchetti?  It use the USB stick LSW3 with firmware LSW3_15_270A_1.32 .  I tried to get the data but I don't get nothig with all the configurations

**Comment by coenieolivier** [comment]
Hi all. I'm new to solar and home assistant. I have an 8Kw Sunsynk interter with what appears to be a newer wifi dongle that does not work with @StephanJoubert's HA add-on integration. It does not allow me to use the dongle's serial number which does not look like the examples provided in the example provided by Stephan. 

My wifi donge serial: E4701xxxQxxx

Is there anyone been able to solve this issue or has a similar issue to this?

**Reply by naboo123** [reply]
I *think* you need to use the inverter's serial number, not the dongle.

**Reply by coenieolivier** [reply]
Thanks for your feedback @naboo123, but I have already tried this and using the inverter's serial is not working either. Is there any info such as HA logs (other than what was shared already) or something I can share that will help shed some light on this so that someone can assist?
Followed documentation as per HACS integration: https://github.com/StephanJoubert/home_assistant_solarman/blob/main/README.md

Option 1: Cannot add wifi dongle serial as it contains alphanumeric characters:
![image](https://user-images.githubusercontent.com/40314327/193556497-6e6511b3-9fc6-41b1-987f-4889a8ee52f9.png)

Option 2: Specified port 0 to allow for "auto-detect", but this does not work as per log error:
![image](https://user-images.githubusercontent.com/40314327/193557153-6be106c3-e7a2-4178-ba48-6a9cb93dc8e4.png)

Option 3: Added Solarman config to configuration.yaml in an attempt to work around the issues experienced in Option 1 and 2:
![image](https://user-images.githubusercontent.com/40314327/193558128-ae18686d-70c7-4436-a092-f8fe1f67bc1a.png)

None of the two options worked as the entity is created (when adding manually to configuration.yaml), but does not update

**Reply by naboo123** [reply]
Apologies - you're right, it is the dongle serial, NOT the inverter. My serial looks the same as the examples provided (ten digit numeric). I am guessing you have a different dongle (I also have a Deye inverter, not Sunsynk - but I understand the hardware to be the same). I have seen on powerforum.co.za that there are other means of integration, and perhaps you can find some help there.

**Reply by coenieolivier** [reply]
Yeah - I was hoping this solution would work as it is much more elegant vs having to install my pie4 next to the inverter with power and cabling, etc. Thank you for your feedback and willingness to help. I've ordered the RS232 cable and will be moving my pie4 to the inverter tonight.

**Comment by emcesmit** [comment]
Good Day Community.

I have been running Home Assistant at my house and on the farm for a few years now. We just replaced our inverters with 3 x Deye inverters. I love HA, but the community around it is what makes the hobby so enjoyable.

How can I connect 3 x Inverters using the solarman integration? As soon as I add a second unit, the values change to "unknown"

**Reply by emcesmit** [reply]
@StephanJoubert Is it possible to have more than one inverter connected at the same time?
